### PR TITLE
Dynamic jwks url by discovery document #417

### PR DESCRIPTION
**Pull request by mabuonomo**
Hi, this PR is linked to issue #417

### CODE REVIEW

File: src/Auth0.php [hunk 1]
```diff
@@ -245,6 +253,11 @@ public function __construct(array $config)
             throw new CoreException('Invalid domain');
         }
 
+        $this->discoveryDocumentUri = $config['discovery_document_uri'] ?? $_ENV['AUTH0_DISCOVERY_DOCUMENT_URI'] ?? null;
```
Comment (joshcanhelp) [inline src/Auth0.php]: Why are we asking for this if we already have the `domain`? The discovery document has a spec-defined URL, `.well-known/openid-configuration`

File: src/Auth0.php [hunk 2]
```diff
@@ -245,6 +253,11 @@ public function __construct(array $config)
...
+        if (empty($this->discoveryDocumentUri)) {
+            throw new CoreException('Invalid discovery document');
```
Comment (joshcanhelp) [inline src/Auth0.php]: This is a breaking change and we're not planning on releasing another major anytime soon. Will need to fallback on the existing URLs if, for some reason, we can't do discovery.

File: src/Auth0.php (hunk 1 above)
Comment (mabuonomo) [reply src/Auth0.php]: You are right, I've forced openid-configuration

**Comment by mabuonomo** [comment]
@joshcanhelp the whitespaces are added by the command "phpcbf", if you want I can restore them.
NB. Also phpcs has many errors

**Comment by joshcanhelp** [comment]
For this project, use `composer phpcbf` to load our PHPCS config file. This is on `master`:

```
~/Sites/php-auth0/auth0 master
❯ composer phpcbf
> "vendor/bin/phpcbf"
................................................ 48 / 48 (100%)

No fixable errors were found
```

**Comment by mabuonomo** [comment]
> For this project, use `composer phpcbf` to load our PHPCS config file. This is on `master`:
> 
> ```
> ~/Sites/php-auth0/auth0 master
> ❯ composer phpcbf
> > "vendor/bin/phpcbf"
> ................................................ 48 / 48 (100%)
> 
> No fixable errors were found
> ```
> 
> PHPCS definitely needs some work here but that should happen in different PR, not this one.

same result on my branch, maybe phpcbf not fix (and check) the spaces.
I've reverted spaces

**Comment by joshcanhelp** [comment]
@mabuonomo - Just FYI ... we're doing some work with the JWKS fetching in #426 that should address your custom JWKS issue. We'll add discovery at some point but that will probably come with a new `Auth0` login helper class.

**Comment by joshcanhelp** [comment]
Addressed in #426

**Pull request closed without merging**